test.js

```diff
@@ -515,13 +516,69 @@ testParseGenerate('no clientId with 3.1.1', {
 
 testParseGenerateDefaults('default connect', {
   cmd: 'connect',
+  retain: false,
+  qos: 0,
+  dup: false,
+  length: 16,
+  topic: null,
+  payload: null,
+  protocolId: 'MQTT',
+  protocolVersion: 4,
+  clean: true,
+  keepalive: 0,
```

Comment:
this tests had this values omitted on purpose for the generation tests, as it tests the defaults are there. It's not equivalent at the previous one.

Reply:
@mcollina, yes, sorry I looked at the test as doing both a Parse and Generate and didn't twig about the defaults (the clue was in the name!). I was after a two way test so I've created a new testParseAndGenerate to do that for my new tests.

On the existing test I have added an additional field so it takes the partial packet object, generates the mqtt bytes, and then parses those to create the fully populated packet object. I believe this now fully covers the test that was previously being done but with the added benefit of checking the generated object is exactly as expected.